docs/01_nodeos/02_usage/03_development-environment/10_local-single-node-testnet-consensus.md

```diff
@@ -0,0 +1,199 @@
+---
+content_title: Local Single-Node Testnet With Consensus Protocol
+link_text: Local Single-Node Testnet With Consensus Protocol
+---
+
+## Goal
+
+This section describes how to set up a single-node blockchain configuration running on a single host with consensus protocol enabled.  This is referred to as a _**single host, single-node testnet with consensus**_.  We will set up one node on your local computer and have it produce blocks. The following diagram depicts the desired single host testnet.
+
+![Single host single node testnet](single-host-single-node-testnet.png)
+
+`cleos` is used to interact with wallets, and send actions to the blockchain. `keosd` performs wallet management, including digital signing. If not started explicitly, `keosd` is started by `cleos` by default.
+
+## Before you begin
+
+* [Install the EOSIO software](../../../00_install/index.md) before starting this section.
+* It is assumed that `nodeos`, `cleos`, and `keosd` are accessible through the path. * Know how to pass [Nodeos options](../../02_usage/00_nodeos-options.md) to enable or disable functionality.
+
+## Steps
+
+Open one "terminal" window and perform the following steps:
+
+1. [Add the development key to the wallet](#1-add-the-development-key-to-the-wallet)
+2. [Start the Producer Node](#2-start-the-producer-node)
+3. [Preactivate Protocol Features](#3-preactivate-protocol-features)
+4. [Build System Smart Contracts](#4-build-system-smart-contracts)
+5. [Install eosio.boot System Contract](#5-install-eosio.boot-system-contract)
+6. [Activate The Remaining Protocol Features](#6-activate-the-remaining-protocol-features)
+7. [Install eosio.bios System Contract](#7-install-eosio.bios-system-contract)
+
+### 1. Add the development key to the wallet
+
+Execute the following at the command prompt:
```

Comment:
this is a how to, it implies user knows more than just basics. this is becoming too much now.

Reply:
Ok it's already mentioned in the step title to add the dev key, so that's sufficient. Btw, despite the title this is not a short how-to as in doing one or two quick steps. It's a short tutorial on how to deploy a local testnet for the first time. Most readers will still be learning the basics most likely. Therefore, in general please be open to add short one-sentence explainers whenever readers perform a step. It's better to be redundant w/o assuming too much on the reader's knowledge in order to impact a wider audience.